src/contentmanager.cpp

```diff
@@ -61,6 +61,12 @@ ContentManager::ContentManager(Library* library, kiwix::Downloader* downloader,
     connect(this, &ContentManager::filterParamsChanged, this, &ContentManager::updateLibrary);
     connect(this, &ContentManager::booksChanged, this, [=]() {
         const auto nBookList = getBooksList();
+        std::call_once(zeroBooksInLocalFlag, [=]() {
+            if (nBookList.size() == 0) {
+                setLocal(false);
+                emit(catalogChanged(isLocal()));
+            }
+        });
```

Comment:
It seems a bit odd to have a call_once in a action being triggered a lot of time.
Setting the catalog local or remote is part of initialization and it should be put here.
Probably just after we load the local library at init.

Reply:
Done.